pkg/plugin/v1/api.go

```diff
@@ -0,0 +1,173 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/pkg/model/resource"
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/pkg/plugin/scaffold"
+
+	"github.com/joelanford/helm-operator/pkg/internal/kubebuilder/cmdutil"
+	"github.com/joelanford/helm-operator/pkg/plugin/internal/chartutil"
+	"github.com/joelanford/helm-operator/pkg/plugin/v1/scaffolds"
+)
+
+type createAPIPlugin struct {
+	config *config.Config
+
+	createOptions chartutil.CreateOptions
+	gvk           schema.GroupVersionKind
+}
+
+var (
+	_ plugin.CreateAPI   = &createAPIPlugin{}
+	_ cmdutil.RunOptions = &createAPIPlugin{}
+)
+
+func (p createAPIPlugin) UpdateContext(ctx *plugin.Context) {
+	ctx.Description = `Scaffold a Kubernetes API that is backed by a Helm chart.
+`
+	ctx.Examples = fmt.Sprintf(`  $ %s create api \
+      --group=apps --version=v1alpha1 \
+      --kind=AppService
+
+  $ %s create api \
+      --group=apps --version=v1alpha1 \
+      --kind=AppService \
+      --helm-chart=myrepo/app
+
+  $ %s create api \
+      --helm-chart=myrepo/app
+
+  $ %s create api \
+      --helm-chart=myrepo/app \
+      --helm-chart-version=1.2.3
+
+  $ %s create api \
+      --helm-chart=app \
+      --helm-chart-repo=https://charts.mycompany.com/
+
+  $ %s create api \
+      --helm-chart=app \
+      --helm-chart-repo=https://charts.mycompany.com/ \
+      --helm-chart-version=1.2.3
+
+  $ %s create api \
+      --helm-chart=/path/to/local/chart-directories/app/
+
+  $ %s create api \
+      --helm-chart=/path/to/local/chart-archives/app-1.2.3.tgz
+`,
+		ctx.CommandName,
+		ctx.CommandName,
+		ctx.CommandName,
+		ctx.CommandName,
+		ctx.CommandName,
+		ctx.CommandName,
+		ctx.CommandName,
+		ctx.CommandName,
+	)
+}
+
+const (
+	group            = "group"
+	version          = "version"
+	kind             = "kind"
+	helmChart        = "helm-chart"
+	helmChartRepo    = "helm-chart-repo"
+	helmChartVerison = "helm-chart-version"
+
+	crdVersion        = "crd-version"
+	crdVersionV1      = "v1"
+	crdVersionV1beta1 = "v1beta1"
+)
+
+func (p *createAPIPlugin) BindFlags(fs *pflag.FlagSet) {
+	p.createOptions = chartutil.CreateOptions{}
+	fs.SortFlags = false
+
+	fs.StringVar(&p.createOptions.GVK.Group, group, "", "resource Group")
+	fs.StringVar(&p.createOptions.GVK.Version, version, "", "resource Version")
+	fs.StringVar(&p.createOptions.GVK.Kind, kind, "", "resource Kind")
+
+	fs.StringVar(&p.createOptions.Chart, helmChart, "", "helm chart")
```

Comment:
```suggestion
	fs.StringVar(&p.createOptions.Chart, helmChart, "", "helm chart (<URL>, <repo>/<name>, or local path)")
```

Reply:
for the api flag ˆ. See: https://github.com/joelanford/helm-operator/pull/45/files#r446512099